js/pages/characterizations/components/characterizations/characterization-view-edit.js

```diff
@@ -189,6 +189,9 @@ define([
             // Next check to see that a characterization with this name does not already exist
             // in the database. Also pass the id so we can make sure that the current characterization is excluded in this check.
             try {
+                if (!constants.isValidName(this.design().name())) {
```

Comment:
In general, best practices tells the good function should have a single entry point and single out.
Regarding this case it's better to use the following sample:
```
if (isValidName(someName)) {
  ...
}
```